src/Common/AzureADWindowsAuthenticator/Program.cs

```diff
@@ -22,7 +23,20 @@ public static void Main(string[] args)
         private static async Task<string> GetToken(ParsedArguments parsedArgs)
         {
             var context = new AuthenticationContext(parsedArgs.AzureADAuthorityAddress);
-            var result = await context.AcquireTokenAsync(parsedArgs.AzureADResource, parsedArgs.AzureADClientId, new Uri(parsedArgs.AzureADRedirectAddress), new PlatformParameters(PromptBehavior.Auto), UserIdentifier.AnyUser, parsedArgs.QueryParams);
+            AuthenticationResult result = null;
+            if(!string.IsNullOrEmpty(parsedArgs.UserName) && !string.IsNullOrEmpty(parsedArgs.Password))
+            {
+                // https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/wiki/Acquiring-tokens-with-username-and-password
+                var passwordBytes = Convert.FromBase64String(parsedArgs.Password);
+                var password = Encoding.UTF8.GetString(passwordBytes);
+                // TODO encrypt password and decode here, either AES or MachineKey (but it needs to work across .NET Framework and .NET Core)
```

Comment:
Are either of these TODOs blocking from a security standpoint?

Reply:
If a sniffer is running on the machine the user is already compromised but ideally I would like to use the Windows Data Protection Library just to harden it. I need to see if that is available in the Windows Compatibility NuGet package as this code only runs on Windows OS machines